torch/_dynamo/variables/nn_module.py

```diff
@@ -283,37 +282,53 @@ def get_kwargs(*names):
             bound_args = bound_args.arguments
             return {k: bound_args[k] for k in names}
 
-        def wrap_values(items, getsource=AttrSource):
+        def wrap_values(items, is_parameter=False):
             result = []
             for name, submod in items:
-                # layer.0.foo => layer[0].foo
-                name = re.sub(r"[.]([0-9]+)([.]|$)", r"[\1]\2", name)
```

Comment:
I think we don't need this any more, as if the source is ```layer.0.foo```, it generate the following bytecode:
```
             40 LOAD_FAST                1 (self)
             42 LOAD_ATTR               15 (layer)
             44 LOAD_ATTR               16 (0)
             46 LOAD_ATTR               14 (foo) 
```
When Python executes these bytecodes, ```LOAD_ATTR``` calls ```getattr```. However, PyTorch nn module override [```__getattr__```](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1574), so it produces same output as ```layer[0].foo```.